lightning-background-processor/src/lib.rs

```diff
@@ -89,31 +89,82 @@ const FIRST_NETWORK_PRUNE_TIMER: u64 = 60;
 #[cfg(test)]
 const FIRST_NETWORK_PRUNE_TIMER: u64 = 1;
 
+/// Either [`P2PGossipSync`] or [`RapidGossipSync`].
+pub enum GossipSync<
+	P: Deref<Target = P2PGossipSync<G, A, L>>,
+	R: Deref<Target = RapidGossipSync<G, L>>,
+	G: Deref<Target = NetworkGraph<L>>,
+	A: Deref,
+	L: Deref,
+>
+where A::Target: chain::Access, L::Target: Logger {
+	/// Gossip sync via the lightning peer-to-peer network as defined by BOLT 7.
+	P2P(P),
+	/// Rapid gossip sync from a trusted server.
+	Rapid(R),
+	/// No gossip sync.
```

Comment:
hm, I actually think an option might convey nullability better than a none, although this definitely takes much less code. However, for people who want to do an initial rapid catch up, but then supplement it with real data as it becomes available, do you think both makes sense?